modules/deployment/iam_code_pipeline.tf

```diff
@@ -0,0 +1,93 @@
+resource "aws_iam_role" "code_pipeline_role" {
+  count              = local.create_code_pipeline_iam ? 1 : 0
+  name               = "code-pipeline-${var.service_name}"
+  path               = local.iam_path
+  assume_role_policy = data.aws_iam_policy_document.allow_code_pipeline_assume[count.index].json
+  tags               = merge(var.tags, {
+    tf_module = basename(path.module)
+  })
+}
+
+data "aws_iam_policy_document" "allow_code_pipeline_assume" {
+  count = local.create_code_pipeline_iam ? 1 : 0
+  statement {
+    actions = ["sts:AssumeRole"]
+    principals {
+      type        = "Service"
+      identifiers = ["codepipeline.amazonaws.com"]
+    }
+  }
+}
+
+resource "aws_iam_policy" "code_pipeline" {
+  count  = local.create_code_pipeline_iam ? 1 : 0
+  name   = "deployment-pipeline-${var.service_name}"
+  path   = local.iam_path
+  policy = data.aws_iam_policy_document.code_pipepline_permissions[count.index].json
+}
+
+resource "aws_iam_role_policy_attachment" "code_pipepline_extra" {
+  count      = local.create_code_pipeline_iam ? 1 : 0
+  role       = aws_iam_role.code_pipeline_role[count.index].name
+  policy_arn = aws_iam_policy.code_pipeline[count.index].arn
+}
+
+data "aws_iam_policy_document" "code_pipepline_permissions" {
+  count = local.create_code_pipeline_iam ? 1 : 0
+
+  statement {
+    actions   = ["ecr:DescribeImages"]
+    resources = [data.aws_ecr_repository.this.arn]
+  }
+
+  statement {
+    actions   = ["s3:GetObject", "s3:ListBucket", "s3:PutObject"]
+    resources = [
+      local.artifact_bucket_arn,
+      "${local.artifact_bucket_arn}/*"
+    ]
+  }
+  #  statement {
```

Comment:
commented code....you specify `ecs:*` down below, just remove?

Reply:
I've tried that, but its not documented what's actually required here. It just fails during deployment with a hint 'not enough permissions'.

I guess the permissions are depending on what you actually do here .. fargate/ec2, alb/elb/nlb/service-discovery. But let's address this in another issue?